packages/admin-panel-server/src/routes/dashboardVisualisations/ImportDashboardVisualisationRoute.ts

```diff
@@ -59,11 +60,40 @@ export class ImportDashboardVisualisationRoute extends Route<ImportDashboardVisu
   }
 
   public async buildResponse() {
-    if (!this.req.file) {
+    const { files } = this.req;
+    if (!files) {
       throw new UploadError();
     }
 
-    const { dashboards = [], dashboardRelations = [], ...visualisation } = this.readFileContents();
+    const validatedFiles = Object.entries(readAndValidateFiles(files, importFileSchema));
+
+    const importedVizIds: string[] = [];
+    const errors: [string, string][] = [];
+    for (let i = 0; i < validatedFiles.length; i++) {
+      const [fileName, file] = validatedFiles[i];
+      try {
+        importedVizIds.push(await this.importViz(file));
+      } catch (error: any) {
+        errors.push([fileName, error.message]);
+      }
+    }
```

Comment:
Using this slightly awkward looping process because I wanted to be able to handle when import errors occurred gracefully. With this, we attempt to import all vizes, and if any fail any we return a message saying which failed, but the rest import successfully. Doing it any other way would be tricky, as there's layers of validation logic deeper so we don't always know if an import will successful until we attempt it.